plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -353,6 +372,11 @@ class producer_plugin_impl : public std::enable_shared_from_this<producer_plugin
             return false;
          }
 
+         if (_block_vault_resync.is_pending() && _producers.count( block->producer ) > 0 ) {
+            // Cancel any pending resync from blockvault if we received any blocks from the same logical producer
+            _block_vault_resync.cancel();
```

Comment:
At this point, we have not validated the signature.   This should be moved underneath the `chain.push_block` call and its error handling when we do other book-keeping like conditionally setting `_production_enabled=true`